tests/test_api_build.py

```diff
@@ -1180,21 +1177,13 @@ def test_failed_recipe_leaves_folders(testing_config, testing_workdir):
         api.build(m)
     assert os.path.isdir(m.config.build_folder), 'build folder was removed'
     assert os.listdir(m.config.build_folder), 'build folder has no files'
+
     # make sure that it does not leave lock files, though, as these cause permission errors on
     #    centralized installations
-    any_locks = False
-    locks_list = set()
-    locale.getpreferredencoding(False)
-    for lock in locks:
-        if os.path.isfile(lock.lock_file):
-            any_locks = True
-            dest_path = base64.b64decode(os.path.basename(lock.lock_file))
-            if hasattr(dest_path, "decode"):
-                dest_path = dest_path.decode(errors="backslashreplace")
-            locks_list.add((lock.lock_file, dest_path))
-    assert not any_locks, "\n".join(
+    locks = {lock.lock_file for lock in locks if os.path.isfile(lock.lock_file)}
+    assert not locks, "\n".join(
         "remaining locks:",
```

Comment:
could there me more than one locked `path`

Reply:
supposedly different locks are created depending on what logic is being blocked from simultaneously running but in the failing test there is only ever one lock that fails to be released

Reply:
that is true.. i think if we need to expand this we will need to see what set of locks are there..